File: app/src/main/java/com/example/punchpad2/StaticScrollEditText.java
package com.example.punchpad2;

import android.content.Context;
import android.util.AttributeSet;
import androidx.appcompat.widget.AppCompatEditText;

public class StaticScrollEditText extends AppCompatEditText {

    public StaticScrollEditText(Context context) {
        super(context);
    }

    public StaticScrollEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public StaticScrollEditText(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean requestRectangleOnScreen(android.graphics.Rect rect) {
        // Block automatic scroll to cursor
        return false;
    }
}

File: app/src/main/java/com/stratonotes/StaticScrollEditText.kt
package com.stratonotes

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatEditText

class StaticScrollEditText : AppCompatEditText {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int)
            : super(context, attrs, defStyleAttr)

    override fun requestRectangleOnScreen(rect: android.graphics.Rect): Boolean {
        // Prevent EditText from auto-scrolling when focused
        return false
    }
}
